documentation/source/building.rst

```diff
@@ -273,3 +273,14 @@ Additional Environment Variables
     Path to the directory containing the cocotb Makefiles and simulator libraries in the subdirectories
     :file:`lib`, :file:`include`, and :file:`makefiles`.
     You don't normally need to modify this.
+
+.. envvar:: GPI_EXTRA
+
+    A comma-separated list of extra libraries that are dynamically loaded at runtime.
+    One can also specify a custom entry point after ``:``.
```

Comment:
Continuing on my comment from before, it's important to note here that these entry functions are run *before* elaboration. One can't access HDL objects at this time, but they can register system functions that are required to be linked at elaboration.

Reply:
```suggestion
    A comma-separated list of extra libraries that are dynamically loaded at runtime.
    A function from each of these libraries will be called prior to elaboration, allowing these libraries to register
    system functions and callbacks. Note that HDL objects cannot be accessed at this time.
    The function defaults to ``{library_name}_entry_point``, but a custom name can be specified using a ``:``.
```